cache/cache.go

```diff
@@ -139,40 +162,163 @@ func (c *Cache[T]) WriteThruLookup(name string, primaryLookup Func[T]) (T, error
 // Where nil, false indicates a cache miss or that the value is expired and should
 // be refreshed.
 func (c *Cache[T]) Lookup(name string) (T, bool) {
+	now := time.Now().UTC()
+
 	c.mu.RLock()
 	defer c.mu.RUnlock()
 
-	return c.lookup(name)
+	if c.isStopped() {
+		panic("cache is stopped")
+	}
+
+	return c.lookup(name, now)
+}
+
+// lookup is the internal implementation of Lookup. Callers are responsible for
+// acquring a lock and checking whether the cache is stopped.
+func (c *Cache[T]) lookup(name string, now time.Time) (T, bool) {
+	v, ok := c.data[name]
+	if !ok || v.expiresAt.Before(now) {
+		var zeroV T
+		return zeroV, false
+	}
+	return v.value, true
 }
 
 // Set saves the current value of an object in the cache, with the supplied
 // durintion until the object expires.
-func (c *Cache[T]) Set(name string, object T) error {
+func (c *Cache[T]) Set(name string, object T) {
+	now := time.Now().UTC()
+
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	if c.isStopped() {
+		panic("cache is stopped")
+	}
+
+	c.set(name, object, now)
+}
+
+func (c *Cache[T]) set(name string, object T, now time.Time) {
+	// Calculate expiration after acquiring a lock. The item is valid from when
+	// insertion started, not when insertion finishes.
+	exp := now.Add(c.expireAfter)
+
+	node, ok := c.data[name]
```

Comment:
There is a bug here - if the value is found, the node that is being pointed to is reused but not _moved_ in the list. 

This actually breaks the linked list chain an keeps it at it's current spot, blocking further cleanup.

Reply:
Addressed, with many more tests.